consensus/hotstuff/committees/consensus_committee.go

```diff
@@ -90,202 +138,202 @@ func (c *Consensus) Identity(blockID flow.Identifier, nodeID flow.Identifier) (*
-	//
-	epochs := c.state.Final().Epochs()
-	next := epochs.Next()
-
-	// TMP: EMERGENCY EPOCH CHAIN CONTINUATION [EECC]
-	//
-	// If we reach this code-path, it means we are about to propose or vote
-	// for the first block in the next epoch. If that epoch has not been
-	// committed or set up, rather than stopping consensus, this intervention
-	// will create a new fallback leader selection for the next epoch containing
-	// 6 months worth of views, so that consensus will have leaders specified
-	// for the duration of the current spork, without any epoch transitions.
-	//
-	_, err = next.DKG() // either of the following errors indicates that we have transitioned into EECC
-	if errors.Is(err, protocol.ErrEpochNotCommitted) || errors.Is(err, protocol.ErrNextEpochNotSetup) {
-		current := epochs.Current()
-
-		currentCounter, err := current.Counter()
-		if err != nil {
-			return flow.ZeroID, fmt.Errorf("could not get next epoch currentCounter: %w", err)
-		}
-		identities, err := current.InitialIdentities()
-		if err != nil {
-			return flow.ZeroID, fmt.Errorf("could not get epoch initial identities: %w", err)
-		}
-		// Get the random source
-		// CAUTION: this is re-using the same leader selection random source from the now-ending epoch
-		randomSeed, err := current.RandomSource()
-		if err != nil {
-			return flow.ZeroID, fmt.Errorf("could not get epoch seed: %w", err)
-		}
-		currentFinalView, err := current.FinalView()
-		if err != nil {
-			return flow.ZeroID, fmt.Errorf("could not get epoch first view: %w", err)
-		}
-
-		// we will inject a fallback leader selection in place of the next epoch
-		counter := currentCounter + 1
-		// the fallback leader selection begins after the final view of the current epoch
-		firstView := currentFinalView + 1
-
-		// create random number generator from the seed and customizer
-		rng, err := seed.PRGFromRandomSource(randomSeed, seed.ProtocolConsensusLeaderSelection)
-		if err != nil {
-			return flow.ZeroID, fmt.Errorf("could not create rng from seed: %w", err)
-		}
+	return epochInfo.initialCommittee.Filter(selector), nil
+}
 
-		selection, err := leader.ComputeLeaderSelection(
-			firstView,
-			rng,
-			int(firstView+leader.EstimatedSixMonthOfViews), // the fallback epoch lasts until the next spork
-			identities.Filter(filter.IsVotingConsensusCommitteeMember),
-		)
-		if err != nil {
-			return flow.ZeroID, fmt.Errorf("could not compute epoch fallback leader selection: %w", err)
-		}
-		c.mu.Lock()
-		c.leaders[counter] = selection
-		c.mu.Unlock()
-		return selection.LeaderForView(view)
-	}
+// IdentityByEpoch returns the identity for the given node ID, in the epoch which
+// contains the given view.
+//
+// Error returns:
+//   * ErrViewForUnknownEpoch if no committed epoch containing the given view is known.
+//     This is an expected error and must be handled.
+//   * unspecific error in case of unexpected problems and bugs
```

Comment:
`InvalidSignerError` should be added as an expected error:
```suggestion
// During normal operations, the following errors are expected:
//   * model.ErrViewForUnknownEpoch if no committed epoch containing the given view is known.
//     This is an expected error and must be handled.
//   * model.InvalidSignerError if nodeID was not listed by the Epoch Setup event as an authorized consensus participants
```

Reply:
added